src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -786,11 +786,16 @@ class CreateTableStatementSegment(BaseSegment):
             # Columns and comment syntax:
             Sequence(
                 Bracketed(
-                    Delimited(
+                    # Manually rebuild Delimited.
+                    # Delimited breaks nested datatypes
+                    # (Comma splits angle bracket blocks)
+                    AnyNumberOf(
                         Sequence(
                             Ref("ColumnDefinitionSegment"),
                             Ref("CommentGrammar", optional=True),
+                            Ref("CommaSegment", optional=True),
                         ),
+                        min_times=1,
```

Comment:
A better option might be:

```py
                    Sequence(
                        Ref("ColumnDefinitionSegment"),
                        Ref("CommentGrammar", optional=True),
                    ),
                    AnyNumberOf(
                        Sequence(
                            Ref("CommaSegment"),
                            Ref("ColumnDefinitionSegment"),
                            Ref("CommentGrammar", optional=True),
                        ),
                    ),
```

This avoid allowing a trailing comma at the end.

Reply:
I appreciate your tips, you're totally right, that makes a whole lot more sense.
While on it, I will add test cases as well.
I will open a new PR - sorry for any confusion caused, it's my first time contributing.

Reply:
No need to open a new PR. Could just update this branch and let me know to re-review once ready. Normal and expected part of PR workflows: https://github.com/sqlfluff/sqlfluff/wiki/Git-Guide#actioning-feedback-from-pull-requests.

This was was a good change. Just think it needed a slight tweak and a test case.

Contributions are VERY much appreciated! And we're happy to help all contributors (particularly first time ones!) through the process.